### PR TITLE
Review hooks

### PR DESCRIPTION
**Pull request by nsivertsen**

### CODE REVIEW

File: Notebook/Reactjs/learning-hooks/src/useInterval.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const useInterval = (callback, wait) => {
   useEffect(() => {
     //everytime it renders save the latest callback ti the current one
     intervalRef.current = callback;
-  });
+  }, [callback]);
```
Comment (nsivertsen) [inline Notebook/Reactjs/learning-hooks/src/useInterval.js:8]: We only need to save a  new callback when it has changed.

File: Notebook/Reactjs/learning-hooks/src/useScroll.js [hunk 2]
```diff
@@ -1,11 +1,9 @@
 import { useState, useEffect } from "react";
 
 const useScroll = () => {
-  const [scrollVertical, setScrollVertical] = useState();
+  const [scrollVertical, setScrollVertical] = useState(0);
```
Comment (nsivertsen) [inline Notebook/Reactjs/learning-hooks/src/useScroll.js:4]: We initialise to a vertical scroll of 0.

File: Notebook/Reactjs/learning-hooks/src/useScroll.js [hunk 3]
```diff
@@ -1,11 +1,9 @@
...
   const handleVerticaScroll = () => {
-    if (window.pageYOffset !== 0) {
-      setScrollVertical(window.pageYOffset);
-    }
+    setScrollVertical(window.pageYOffset);
```
Comment (nsivertsen) [inline Notebook/Reactjs/learning-hooks/src/useScroll.js:6]: A scroll of 0 is also a valid value.

File: Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js [hunk 4]
```diff
@@ -1,31 +1,28 @@
...
 
+function debounce(callback, wait) {
+  let timeout;
+  return (...args) => {
+    const context = this;
+    clearTimeout(timeout);
+    timeout = setTimeout(() => callback.apply(context, args), wait);
+  };
+}
+
```
Comment (nsivertsen) [inline Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js:11]: This function doesn't depend on the component's state, so it makes sense to move it out. This is also more performant.

File: Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js [hunk 5]
```diff
@@ -1,31 +1,28 @@
...
-      timeout = setTimeout(() => callback.apply(context, args), wait);
-    };
-  }
-
   useEffect(() => {
-    window.addEventListener(
-      "resize",
-      debounce(() => {
-        calcWindowSize();
-      }, 500)
```
Comment (nsivertsen) [inline Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js:27]:
```
      debounce(() => {	      const width = window.innerWidth;
        calcWindowSize();	      setInnerWidth(width);
      }, 500)
```

and 

```
calcWindowSize
```

are two different functions, meaning `removeEventListener` won't register the handler.

File: Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js [hunk 6]
```diff
@@ -1,31 +1,28 @@
...
-    );
+    const calcWindowSize = debounce(() => {
+      const height = window.innerHeight;
+      const width = window.innerWidth;
+      setInnerWidth(width);
+      setInnerHeight(height);
+    }, 500);
+
+    calcWindowSize();
```
Comment (nsivertsen) [inline Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js:24]: We call `calcWindowSize` once initially to get the first value.

**Review by nsivertsen — COMMENTED** [review]
Very nice work! Just a few comments below :)

**Review by lalawuhan — APPROVED** [review]

**Pull request merged**